src/strings/string_printf.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2013 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef STRINGS_STRING_PRINTF_H_
+#define STRINGS_STRING_PRINTF_H_
+
+#include <stdarg.h>   // va_list
+
+#include <string>
+
+namespace strings {
+
+// Return a C++ string given printf-like input.
+std::string StringPrintf(const char* format, ...)
+    __attribute__((format(printf, 1, 2)))
+    __attribute__((warn_unused_result));
+
+// Return a C++ string given vprintf-like input.
+std::string StringPrintV(const char* format, va_list ap)
+    __attribute__((format(printf, 1, 0)))
+    __attribute__((warn_unused_result));
+
+// Store result into a supplied string and return it.
+const std::string& SStringPrintf(std::string* dst, const char* format, ...)
```

Comment:
It is used in the unit tests but otherwise we don't really need it, will review the tests to see if it can be dropped and achieve the same level of testing.